src/main/java/com/iexec/worker/config/BlockchainAdapterConfigurationService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.config;
+
+import com.iexec.common.config.PublicChainConfig;
+import com.iexec.worker.feign.CustomBlockchainAdapterClient;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+
+import static org.web3j.protocol.core.JsonRpc2_0Web3j.DEFAULT_BLOCK_TIME;
+
+@Slf4j
+@Service
+public class BlockchainAdapterConfigurationService {
+    private final PublicChainConfig publicChainConfig;
+
+    public BlockchainAdapterConfigurationService(CustomBlockchainAdapterClient customBlockchainAdapterClient) {
```

Comment:
Might add a note with something like "Restart is required for retrieving latest fresh remote value"

Reply:
My concern was "How our client is notified when the remote has a new exposed value"

Reply:
They currently aren't. We could add a scheduled method that looks for updates once in a while if you think this could be an issue.

Reply:
Yes they aren't, a warning message reminding that point is enough for now